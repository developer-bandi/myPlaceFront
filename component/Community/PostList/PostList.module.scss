@import "../../../lib/styles/globalmixin.scss";

.mainBlock {
  width: 1000px;
  margin: 30px auto 100px auto;
  min-height: calc(100vh - 342px);
  @media screen and (max-width: 1020px) {
    width: calc(100% - 40px);
    margin: 30px 20px 40px 20px;
  }
}

.rightBlock {
  display: flex;
  margin-left: auto;
  @media screen and (max-width: 767px) {
    width: 100%;
    margin: 20px 0px 0px 0px;
  }
}

.title {
  margin: 0px 0px 40px 0px;
}

.searchBlock {
  display: flex;
  align-items: center;
  border: 1px solid #c9c9c9;
  border-radius: 20px;
  height: 35px;
  width: 260px;
  margin: 0px 10px 0px 0px;
  &:focus {
    border: 1px solid #00aaff;
  }
  @media screen and (max-width: 767px) {
    width: calc(100% - 125px);
  }
}

.searchInput {
  width: 200px;
  height: 30px;
  font-size: 20px;
  @media screen and (max-width: 767px) {
    width: 140px;
  }
}

.searchButton {
  margin: 0px 5px 0px 10px;
  display: flex;
  align-items: center;
}

.initializeSearch {
  border-radius: 5px;
  padding: 3px;
  cursor: pointer;
  background-color: gray;
  color: white;
  width: 50px;
  @include centerSort();
  margin: 0px 10px 0px 0px;
  &:hover {
    opacity: 0.8;
  }
}

.sortBlock {
  padding: 0 0 20px 0;
  display: flex;
  border-bottom: 1.5px solid #e5e5e5;
  align-items: flex-end;
  flex-wrap: wrap;
}

.sortList {
  display: flex;
  width: 200px;
  @media screen and (max-width: 767px) {
    width: 100%;
  }
}

.sortName {
  margin-left: 10px;
  color: gray;
  &:hover {
    color: black;
    font-weight: 700;
  }
}

.selectedSortName {
  margin-left: 10px;
  color: black;
  font-weight: 700;
}

.writeButton {
  background-color: #3e4042;
  color: white;
  padding: 10px;
  border-radius: 5px;
  position: relative;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
}

.loading {
  width: 1000px;
  height: 1000px;
  padding: 300px 0 0 450px;
}

.postBlock {
  width: 1000px;
  border-bottom: 1.5px solid #e5e5e5;
  padding: 20px;
  &:hover {
    background-color: #f8f9fa;
  }
  cursor: pointer;
  @media screen and (max-width: 1020px) {
    width: 100%;
  }
}

.postTopBlock {
  display: flex;
  justify-content: space-between;
}

.postTitle {
  margin: 0 auto 20px 0;
}

.postUser {
  margin-left: 10px;
  color: gray;
}

.postWritedAt {
  margin-left: 10px;
  color: #878f8c;
}

.postContent {
  margin: 0 0 20px 0;
  color: #616465;
  line-height: 30px;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.postBottomBlock {
  display: flex;
}

.postBottomInfo {
  margin-right: 10px;
  display: flex;
  align-items: center;
}

.postIcon {
  margin-right: 3px;
}

.message {
  text-align: center;
  margin-top: 100px;
  @include error();
}

.loadingBlock {
  position: relative;
  width: 100%;
  height: 700px;
}
