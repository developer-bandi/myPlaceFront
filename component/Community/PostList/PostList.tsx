import styles from "./PostList.module.scss";
import {FcLike} from "react-icons/fc";
import {GrView} from "react-icons/gr";
import {BiCommentDetail, BiSearchAlt2} from "react-icons/bi";
import {RefObject, SetStateAction} from "react";
import Image from "next/image";
import {setDateLatest} from "../../../lib/commonFn/date";
import PageNation from "../../Common/PageNation/PageNation";
import {postListState} from "./PostListHook";

interface PostListProps {
  page: number;
  postList: postListState;
  changeSort: (sort: string) => Promise<void>;
  changePage: (page: number) => Promise<void>;
  selectedSort: string;
  searchRef: RefObject<HTMLInputElement>;
  searchPost: (e: {key?: string; type: string}) => Promise<void>;
  initializeSearch: () => void;
  moveWritePage: () => void;
  movePostDetailPage: (id: number) => void;
}

const PostList = ({
  page,
  postList,
  changeSort,
  changePage,
  selectedSort,
  searchRef,
  searchPost,
  initializeSearch,
  moveWritePage,
  movePostDetailPage,
}: PostListProps) => {
  if (postList.loading) {
    return (
      <main className={styles.mainBlock}>
        <h1 className={styles.title}>커뮤니티</h1>
        <div className={styles.sortBlock}>
          <ul className={styles.sortList}>
            {[
              {kr: "최신순", en: "createdAt"},
              {kr: "좋아요순", en: "likeCount"},
              {kr: "조회수순", en: "viewCount"},
            ].map((sortNameObj) => {
              return (
                <li
                  className={
                    selectedSort === sortNameObj.en
                      ? styles.selectedSortName
                      : styles.sortName
                  }
                  onClick={() => {
                    changeSort(sortNameObj.en);
                  }}
                  key={sortNameObj.kr}
                  data-testid={`${sortNameObj.en}sortButton`}
                >
                  {sortNameObj.kr}
                </li>
              );
            })}
          </ul>
          <div className={styles.rightBlock}>
            <div className={styles.searchBlock}>
              <button className={styles.searchButton} onClick={searchPost}>
                <BiSearchAlt2 size={"20"} />
              </button>
              <input
                className={styles.searchInput}
                ref={searchRef}
                onKeyPress={searchPost}
              />
            </div>
            <button
              onClick={initializeSearch}
              className={styles.initializeSearch}
            >
              초기화
            </button>
            <button
              className={styles.writeButton}
              onClick={moveWritePage}
              data-testid="writeButton"
            >
              글쓰기
            </button>
          </div>
        </div>
        <div className={styles.loadingBlock}>
          <Image
            src="/searchResultLoading.gif"
            alt="searchImg"
            layout="fill"
            objectFit="contain"
            priority={true}
            data-testid="loading"
          ></Image>
        </div>
      </main>
    );
  } else if (postList.error) {
    return (
      <main className={styles.mainBlock}>
        <h1>커뮤니티</h1>
        <p className={styles.message}>서버에 에러가 발생하였습니다</p>
      </main>
    );
  } else if (postList.content?.count === 0) {
    return (
      <main className={styles.mainBlock}>
        <h1>커뮤니티</h1>
        <div className={styles.sortBlock}>
          <ul className={styles.sortList}>
            {[
              {kr: "최신순", en: "createdAt"},
              {kr: "좋아요순", en: "likeCount"},
              {kr: "조회수순", en: "viewCount"},
            ].map((sortNameObj) => {
              return (
                <li
                  className={
                    selectedSort === sortNameObj.en
                      ? styles.selectedSortName
                      : styles.sortName
                  }
                  onClick={() => {
                    changeSort(sortNameObj.en);
                  }}
                  key={sortNameObj.kr}
                  data-testid={`${sortNameObj.en}sortButton`}
                >
                  {sortNameObj.kr}
                </li>
              );
            })}
          </ul>
          <div className={styles.rightBlock}>
            <div className={styles.searchBlock}>
              <button className={styles.searchButton} onClick={searchPost}>
                <BiSearchAlt2 size={"20"} />
              </button>
              <input
                className={styles.searchInput}
                ref={searchRef}
                onKeyPress={searchPost}
              />
            </div>
            <button
              onClick={initializeSearch}
              className={styles.initializeSearch}
            >
              초기화
            </button>
            <button
              className={styles.writeButton}
              onClick={moveWritePage}
              data-testid="writeButton"
            >
              글쓰기
            </button>
          </div>
        </div>
        <p className={styles.message}>포스트가 존재하지 않습니다</p>
      </main>
    );
  } else {
    return (
      <main className={styles.mainBlock}>
        <h1 className={styles.title}>커뮤니티</h1>
        <div className={styles.sortBlock}>
          <ul className={styles.sortList}>
            {[
              {kr: "최신순", en: "createdAt"},
              {kr: "좋아요순", en: "likeCount"},
              {kr: "조회수순", en: "viewCount"},
            ].map((sortNameObj) => {
              return (
                <li
                  className={
                    selectedSort === sortNameObj.en
                      ? styles.selectedSortName
                      : styles.sortName
                  }
                  onClick={() => {
                    changeSort(sortNameObj.en);
                  }}
                  key={sortNameObj.kr}
                  data-testid={`${sortNameObj.en}sortButton`}
                >
                  {sortNameObj.kr}
                </li>
              );
            })}
          </ul>
          <div className={styles.rightBlock}>
            <div className={styles.searchBlock}>
              <button
                className={styles.searchButton}
                onClick={searchPost}
                data-testid="searchButton"
              >
                <BiSearchAlt2 size={"20"} />
              </button>
              <input
                className={styles.searchInput}
                ref={searchRef}
                onKeyPress={searchPost}
              />
            </div>
            <button
              onClick={initializeSearch}
              className={styles.initializeSearch}
              data-testid="initializeButton"
            >
              초기화
            </button>
            <button
              className={styles.writeButton}
              onClick={moveWritePage}
              data-testid="writeButton"
            >
              글쓰기
            </button>
          </div>
        </div>

        <div className={styles.postListBlock}>
          {postList.content !== undefined &&
            postList.content.rows.map((post, index) => {
              return (
                <article
                  className={styles.postBlock}
                  onClick={() => movePostDetailPage(post.id)}
                  data-testid={`articleButton${index}`}
                  key={index}
                >
                  <div className={styles.postTopBlock}>
                    <h3 className={styles.postTitle}>{post.title}</h3>
                    <div className={styles.postUser}>{post.nickname}</div>
                    <div className={styles.postWritedAt}>
                      {setDateLatest(post.createdAt)}
                    </div>
                  </div>
                  <p className={styles.postContent}>{post.content}</p>
                  <ul className={styles.postBottomBlock}>
                    <li className={styles.postBottomInfo}>
                      <GrView className={styles.postIcon} />
                      {post.viewCount}
                    </li>
                    <li className={styles.postBottomInfo}>
                      <FcLike className={styles.postIcon} />
                      {post.postlikecount}
                    </li>
                    <li className={styles.postBottomInfo}>
                      <BiCommentDetail className={styles.postIcon} />
                      {post.comment}
                    </li>
                  </ul>
                </article>
              );
            })}
        </div>
        <PageNation
          page={page}
          changePage={changePage}
          totalCount={postList.content?.count as number}
          addStyle={"margin"}
        />
      </main>
    );
  }
};

export default PostList;
