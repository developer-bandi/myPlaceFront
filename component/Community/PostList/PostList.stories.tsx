import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import PostList from "./PostList";
import { INITIAL_VIEWPORTS } from "@storybook/addon-viewport";

export default {
  title: "Community/PostList",
  component: PostList,
  parameters: {
    viewport: {
      viewports: INITIAL_VIEWPORTS,
      defaultViewport: "desktop",
    },
  },
} as ComponentMeta<typeof PostList>;

const Template: ComponentStory<typeof PostList> = (arg) => (
  <PostList {...arg} />
);

const content = {
  count: 10,
  rows: new Array(10).fill(0).map((_, index) => {
    return {
      id: index + 1,
      title: "테스트 제목",
      content: "테스트 내용입니다",
      viewCount: 10,
      createdAt: new Date("2022-11-11").toDateString(),
      nickname: "place",
      comment: 0,
      postlikecount: 1,
    };
  }),
};

export const isLoading = Template.bind({});
isLoading.args = {
  postList: {
    loading: true,
    error: false,
  },
};

export const isError = Template.bind({});
isError.args = {
  postList: {
    loading: false,
    error: true,
  },
};

export const isNoResult = Template.bind({});
isNoResult.args = {
  postList: {
    content: { count: 0, rows: [] },
    loading: false,
    error: false,
  },
};

export const isNormal = Template.bind({});
isNormal.args = {
  postList: {
    content,
    loading: false,
    error: false,
  },
  page: 1,
};
