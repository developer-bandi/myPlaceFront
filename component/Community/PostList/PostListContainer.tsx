import PostList from "./PostList";
import usePostList, {postListContent} from "./PostListHook";
interface PostListContainerProps {
  serverSideData: postListContent;
}
const PostListContainer = ({serverSideData}: PostListContainerProps) => {
  const {
    page,
    postList,
    changeSort,
    changePage,
    selectedSort,
    searchRef,
    searchPost,
    initializeSearch,
    moveWritePage,
    movePostDetailPage,
  } = usePostList(serverSideData);

  return (
    <PostList
      page={page}
      postList={postList}
      changeSort={changeSort}
      changePage={changePage}
      selectedSort={selectedSort}
      searchRef={searchRef}
      searchPost={searchPost}
      initializeSearch={initializeSearch}
      moveWritePage={moveWritePage}
      movePostDetailPage={movePostDetailPage}
    />
  );
};

export default PostListContainer;
