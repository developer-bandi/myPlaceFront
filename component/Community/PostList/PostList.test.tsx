import {
  act,
  cleanup,
  fireEvent,
  render,
  renderHook,
  screen,
  waitFor,
} from "@testing-library/react";
import axios from "axios";
import { Provider } from "react-redux";
import usePostList from "./PostListHook";
import configureMockStore from "redux-mock-store";
import { useRouter } from "next/router";
import PostList from "./PostList";
import { ReactNode } from "react";

jest.mock("axios");
jest.mock("next/router", () => ({
  __esModule: true,
  useRouter: jest.fn(),
}));

const axiosMock = axios as jest.Mocked<typeof axios>;
const postListMock = {
  count: 1,
  rows: [
    {
      id: 1,
      title: "testTitle",
      content: "testContent",
      viewCount: 1,
      createdAt: "testCreatedAt",
      nickname: "testNickname",
      comment: 1,
      postlikecount: 1,
    },
  ],
};

// describe("PostList Presentational Test", () => {
//   it("로딩중", () => {
//     const utils = render(
//       <PostList
//         page={1}
//         postList={{loading: true, error: false}}
//         changeSort={jest.fn()}
//         changePage={jest.fn()}
//         selectedSort={"char"}
//         searchRef={{current: null}}
//         searchPost={jest.fn()}
//         initializeSearch={jest.fn()}
//         moveWritePage={jest.fn()}
//         movePostDetailPage={jest.fn()}
//       />
//     );
//     expect(utils.container).toMatchSnapshot();
//     screen.getByTestId("loading");
//   });
//   it("에러 발생", () => {
//     const utils = render(
//       <PostList
//         page={1}
//         postList={{loading: false, error: true}}
//         changeSort={jest.fn()}
//         changePage={jest.fn()}
//         selectedSort={"char"}
//         searchRef={{current: null}}
//         searchPost={jest.fn()}
//         initializeSearch={jest.fn()}
//         moveWritePage={jest.fn()}
//         movePostDetailPage={jest.fn()}
//       />
//     );
//     expect(utils.container).toMatchSnapshot();
//     screen.getByText("서버에 에러가 발생하였습니다");
//   });
//   it("정상출력", () => {
//     const changeSortMock = jest.fn();
//     const searchPostMock = jest.fn();
//     const initializeSearchMock = jest.fn();
//     const moveWritePageMock = jest.fn();
//     const movePostDetailPageMock = jest.fn();

//     const utils = render(
//       <PostList
//         page={1}
//         postList={{content: postListMock, loading: false, error: false}}
//         changeSort={changeSortMock}
//         changePage={jest.fn()}
//         selectedSort={"char"}
//         searchRef={{current: null}}
//         searchPost={searchPostMock}
//         initializeSearch={initializeSearchMock}
//         moveWritePage={moveWritePageMock}
//         movePostDetailPage={movePostDetailPageMock}
//       />
//     );
//     expect(utils.container).toMatchSnapshot();
//     fireEvent.click(screen.getByTestId("searchButton"));
//     fireEvent.click(screen.getByTestId("initializeButton"));
//     fireEvent.click(screen.getByTestId("createdAtsortButton"));
//     fireEvent.click(screen.getByTestId("writeButton"));
//     fireEvent.click(screen.getByTestId("articleButton0"));
//     expect(searchPostMock).toBeCalledTimes(1);
//     expect(initializeSearchMock).toBeCalledTimes(1);
//     expect(changeSortMock).toBeCalledTimes(1);
//     expect(moveWritePageMock).toBeCalledTimes(1);
//     expect(movePostDetailPageMock).toBeCalledTimes(1);
//   });
// });

describe("PostList Hook test", () => {
  afterEach(cleanup);

  let loginStatus = false;
  const mockStore = configureMockStore();
  const loginMockStore = mockStore({
    userLogin: {
      content: { id: 1, nickname: "test" },
      loading: false,
      error: false,
    },
  });
  const logoutMockStore = configureMockStore()({
    userLogin: {
      loading: false,
      error: false,
    },
  });
  const wrapper = ({ children }: { children: ReactNode }) => (
    <Provider store={loginStatus ? loginMockStore : logoutMockStore}>
      {children}
    </Provider>
  );
  window.scrollTo = jest.fn();

  describe("changeSort 함수 검증", () => {
    describe("정상적으로 받아온 경우", () => {
      it("검색 키워드가 있는 경우", async () => {
        axiosMock.get.mockImplementation(() =>
          Promise.resolve({ status: 200, data: postListMock })
        );
        const { result } = renderHook(() => usePostList(postListMock), {
          wrapper,
          initialProps: {
            login: false,
          },
        });
        act(() => {
          result.current.setKeyword("test");
        });
        await act(async () => {
          await result.current.changeSort("likeCount");
        });
        expect(result.current.postList).toStrictEqual({
          content: postListMock,
          loading: false,
          error: false,
        });
        expect(result.current.selectedSort).toBe("likeCount");
        expect(axiosMock.get.mock.calls[0][0]).toBe(
          `${
            process.env.NEXT_PUBLIC_SERVER_DOMAIN
          }/post/search?keyword=${"test"}&page=${1}&order=${"likeCount"}`
        );
        expect(result.current.page).toBe(1);
      });
      it("검색 키워드가 없는 경우", async () => {
        axiosMock.get.mockImplementation(() =>
          Promise.resolve({ status: 200, data: postListMock })
        );
        const { result } = renderHook(() => usePostList(postListMock), {
          wrapper,
          initialProps: {
            login: false,
          },
        });
        await act(async () => {
          await result.current.changeSort("likeCount");
        });
        expect(result.current.postList).toStrictEqual({
          content: postListMock,
          loading: false,
          error: false,
        });
        expect(result.current.selectedSort).toBe("likeCount");
        expect(axiosMock.get.mock.calls[1][0]).toBe(
          `${
            process.env.NEXT_PUBLIC_SERVER_DOMAIN
          }/post/list?page=${1}&order=${"likeCount"}`
        );
        expect(result.current.page).toBe(1);
      });
    });
    it("에러가 발생한 경우", async () => {
      axiosMock.get.mockImplementation(() =>
        Promise.reject({ status: 500, data: "에러발생" })
      );
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
        initialProps: {
          login: false,
        },
      });
      await act(async () => {
        await result.current.changeSort("likeCount");
      });
      expect(result.current.postList).toStrictEqual({
        loading: false,
        error: true,
      });
    });
  });
  describe("changePage 함수 검증", () => {
    describe("정상적으로 받아온 경우", () => {
      it("검색 키워드가 있는 경우", async () => {
        axiosMock.get.mockImplementation(() =>
          Promise.resolve({ status: 200, data: postListMock })
        );
        const { result } = renderHook(() => usePostList(postListMock), {
          wrapper,
          initialProps: {
            login: false,
          },
        });
        act(() => {
          result.current.setKeyword("test");
        });
        await act(async () => {
          await result.current.changePage(2);
        });
        expect(result.current.postList).toStrictEqual({
          content: postListMock,
          loading: false,
          error: false,
        });
        expect(result.current.selectedSort).toBe("createdAt");
        expect(axiosMock.get.mock.calls[3][0]).toBe(
          `${
            process.env.NEXT_PUBLIC_SERVER_DOMAIN
          }/post/search?keyword=${"test"}&page=${2}&order=${"createdAt"}`
        );
        expect(result.current.page).toBe(2);
      });
      it("검색 키워드가 없는 경우", async () => {
        axiosMock.get.mockImplementation(() =>
          Promise.resolve({ status: 200, data: postListMock })
        );
        const { result } = renderHook(() => usePostList(postListMock), {
          wrapper,
          initialProps: {
            login: false,
          },
        });
        await act(async () => {
          await result.current.changePage(2);
        });
        expect(result.current.postList).toStrictEqual({
          content: postListMock,
          loading: false,
          error: false,
        });
        expect(result.current.selectedSort).toBe("createdAt");
        expect(axiosMock.get.mock.calls[4][0]).toBe(
          `${
            process.env.NEXT_PUBLIC_SERVER_DOMAIN
          }/post/list?page=${2}&order=${"createdAt"}`
        );
        expect(result.current.page).toBe(2);
      });
    });
    it("에러가 발생한 경우", async () => {
      axiosMock.get.mockImplementation(() =>
        Promise.reject({ status: 500, data: "에러발생" })
      );
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
        initialProps: {
          login: false,
        },
      });
      await act(async () => {
        await result.current.changePage(2);
      });
      expect(result.current.postList).toStrictEqual({
        loading: false,
        error: true,
      });
    });
  });

  describe("searchPost 함수 검증", () => {
    it("클릭하여 정상적으로 받아온 경우", async () => {
      axiosMock.get.mockImplementation(() =>
        Promise.resolve({ status: 200, data: postListMock })
      );
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
        initialProps: {
          login: false,
        },
      });
      act(() => {
        result.current.setKeyword("test");
      });
      await act(async () => {
        await result.current.searchPost({ type: "click" });
      });
      expect(result.current.postList).toStrictEqual({
        content: postListMock,
        loading: false,
        error: false,
      });
      expect(result.current.selectedSort).toBe("createdAt");
      expect(result.current.keyword).toBe("");
      expect(result.current.page).toBe(1);
    });
    it("엔터키를 눌러 정상적으로 받아온 경우", async () => {
      axiosMock.get.mockImplementation(() =>
        Promise.resolve({ status: 200, data: postListMock })
      );
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
        initialProps: {
          login: false,
        },
      });
      act(() => {
        result.current.setKeyword("test");
      });
      await act(async () => {
        await result.current.searchPost({ type: "click" });
      });
      expect(result.current.postList).toStrictEqual({
        content: postListMock,
        loading: false,
        error: false,
      });
      expect(result.current.selectedSort).toBe("createdAt");
      expect(result.current.keyword).toBe("");
      expect(result.current.page).toBe(1);
    });
    it("에러가 발생한 경우", async () => {
      axiosMock.get.mockImplementation(() =>
        Promise.reject({ status: 500, data: "에러발생" })
      );
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
      });
      act(() => {
        result.current.setKeyword("test");
      });
      await act(async () => {
        await result.current.searchPost({ type: "keypress", key: "Enter" });
      });
      expect(result.current.postList).toStrictEqual({
        loading: false,
        error: true,
      });
    });
  });

  describe("moveWritePage 함수 검증", () => {
    const mockRouter = {
      push: jest.fn(),
    };
    (useRouter as jest.Mock).mockReturnValue(mockRouter);
    it("로그인 한 경우", async () => {
      loginStatus = true;
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
      });
      act(() => {
        result.current.moveWritePage();
      });
      await waitFor(() => {
        expect(mockRouter.push.mock.calls[0][0]).toBe("/community/writepost");
      });
    });

    it("로그인 하지 않은 경우", async () => {
      loginStatus = false;
      window.alert = jest.fn();
      const { result } = renderHook(() => usePostList(postListMock), {
        wrapper,
      });

      act(() => {
        result.current.moveWritePage();
      });
      await waitFor(() => {
        expect(mockRouter.push.mock.calls[1][0]).toBe("/user/auth/signin");
        expect(window.alert).toHaveBeenCalledWith("로그인이 필요합니다.");
      });
    });
  });
});
