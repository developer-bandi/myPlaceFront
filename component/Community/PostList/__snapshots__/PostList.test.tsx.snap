// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList Presentational Test 로딩중 1`] = `
<div>
  <main
    class="mainBlock"
  >
    <h1
      class="title"
    >
      커뮤니티
    </h1>
    <div
      class="sortBlock"
    >
      <ul
        class="sortList"
      >
        <li
          class="sortName"
          data-testid="createdAtsortButton"
        >
          최신순
        </li>
        <li
          class="sortName"
          data-testid="likeCountsortButton"
        >
          좋아요순
        </li>
        <li
          class="sortName"
          data-testid="viewCountsortButton"
        >
          조회수순
        </li>
      </ul>
      <div
        class="rightBlock"
      >
        <div
          class="searchBlock"
        >
          <button
            class="searchButton"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.023 16.977a35.13 35.13 0 0 1-1.367-1.384c-.372-.378-.596-.653-.596-.653l-2.8-1.337A6.962 6.962 0 0 0 16 9c0-3.859-3.14-7-7-7S2 5.141 2 9s3.14 7 7 7c1.763 0 3.37-.66 4.603-1.739l1.337 2.8s.275.224.653.596c.387.363.896.854 1.384 1.367l1.358 1.392.604.646 2.121-2.121-.646-.604c-.379-.372-.885-.866-1.391-1.36zM9 14c-2.757 0-5-2.243-5-5s2.243-5 5-5 5 2.243 5 5-2.243 5-5 5z"
              />
            </svg>
          </button>
          <input
            class="searchInput"
          />
        </div>
        <button
          class="initializeSearch"
        >
          초기화
        </button>
        <button
          class="writeButton"
          data-testid="writeButton"
        >
          글쓰기
        </button>
      </div>
    </div>
    <div
      class="loadingBlock"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="searchImg"
          data-nimg="fill"
          data-testid="loading"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2FsearchResultLoading.gif&w=3840&q=75"
          srcset="/_next/image?url=%2FsearchResultLoading.gif&w=640&q=75 640w, /_next/image?url=%2FsearchResultLoading.gif&w=750&q=75 750w, /_next/image?url=%2FsearchResultLoading.gif&w=828&q=75 828w, /_next/image?url=%2FsearchResultLoading.gif&w=1080&q=75 1080w, /_next/image?url=%2FsearchResultLoading.gif&w=1200&q=75 1200w, /_next/image?url=%2FsearchResultLoading.gif&w=1920&q=75 1920w, /_next/image?url=%2FsearchResultLoading.gif&w=2048&q=75 2048w, /_next/image?url=%2FsearchResultLoading.gif&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
      </span>
    </div>
  </main>
</div>
`;

exports[`PostList Presentational Test 에러 발생 1`] = `
<div>
  <main
    class="mainBlock"
  >
    <h1>
      커뮤니티
    </h1>
    <p
      class="message"
    >
      서버에 에러가 발생하였습니다
    </p>
  </main>
</div>
`;

exports[`PostList Presentational Test 정상출력 1`] = `
<div>
  <main
    class="mainBlock"
  >
    <h1
      class="title"
    >
      커뮤니티
    </h1>
    <div
      class="sortBlock"
    >
      <ul
        class="sortList"
      >
        <li
          class="sortName"
          data-testid="createdAtsortButton"
        >
          최신순
        </li>
        <li
          class="sortName"
          data-testid="likeCountsortButton"
        >
          좋아요순
        </li>
        <li
          class="sortName"
          data-testid="viewCountsortButton"
        >
          조회수순
        </li>
      </ul>
      <div
        class="rightBlock"
      >
        <div
          class="searchBlock"
        >
          <button
            class="searchButton"
            data-testid="searchButton"
          >
            <svg
              fill="currentColor"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.023 16.977a35.13 35.13 0 0 1-1.367-1.384c-.372-.378-.596-.653-.596-.653l-2.8-1.337A6.962 6.962 0 0 0 16 9c0-3.859-3.14-7-7-7S2 5.141 2 9s3.14 7 7 7c1.763 0 3.37-.66 4.603-1.739l1.337 2.8s.275.224.653.596c.387.363.896.854 1.384 1.367l1.358 1.392.604.646 2.121-2.121-.646-.604c-.379-.372-.885-.866-1.391-1.36zM9 14c-2.757 0-5-2.243-5-5s2.243-5 5-5 5 2.243 5 5-2.243 5-5 5z"
              />
            </svg>
          </button>
          <input
            class="searchInput"
          />
        </div>
        <button
          class="initializeSearch"
          data-testid="initializeButton"
        >
          초기화
        </button>
        <button
          class="writeButton"
          data-testid="writeButton"
        >
          글쓰기
        </button>
      </div>
    </div>
    <div
      class="postListBlock"
    >
      <article
        class="postBlock"
        data-testid="articleButton0"
      >
        <div
          class="postTopBlock"
        >
          <h3
            class="postTitle"
          >
            testTitle
          </h3>
          <div
            class="postUser"
          >
            testNickname
          </div>
          <div
            class="postWritedAt"
          >
            testCreatedAt
          </div>
        </div>
        <p
          class="postContent"
        >
          testContent
        </p>
        <ul
          class="postBottomBlock"
        >
          <li
            class="postBottomInfo"
          >
            <svg
              class="postIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,21 C7,21 1,16 1,12 C1,8 7,3 12,3 C17,3 23,8 23,12 C23,16 17,21 12,21 Z M12,7 C9.23875,7 7,9.23875 7,12 C7,14.76125 9.23875,17 12,17 C14.76125,17 17,14.76125 17,12 C17,9.23875 14.76125,7 12,7 L12,7 Z"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
            1
          </li>
          <li
            class="postBottomInfo"
          >
            <svg
              class="postIcon"
              enable-background="new 0 0 48 48"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              version="1"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34,9c-4.2,0-7.9,2.1-10,5.4C21.9,11.1,18.2,9,14,9C7.4,9,2,14.4,2,21c0,11.9,22,24,22,24s22-12,22-24 C46,14.4,40.6,9,34,9z"
                fill="#F44336"
              />
            </svg>
            1
          </li>
          <li
            class="postBottomInfo"
          >
            <svg
              class="postIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7 7h10v2H7zm0 4h7v2H7z"
              />
              <path
                d="M20 2H4c-1.103 0-2 .897-2 2v18l5.333-4H20c1.103 0 2-.897 2-2V4c0-1.103-.897-2-2-2zm0 14H6.667L4 18V4h16v12z"
              />
            </svg>
            1
          </li>
        </ul>
      </article>
    </div>
    <ul
      class="pageButtonList margin"
    >
      <li>
        <svg
          class="pageIcon ban"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            fill="none"
            points="9 6 15 12 9 18"
            stroke="#000"
            stroke-width="2"
            transform="matrix(-1 0 0 1 24 0)"
          />
        </svg>
      </li>
      <li
        class="pageButton selected"
      >
        1
      </li>
      <li>
        <svg
          class="pageIcon ban"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            fill="none"
            points="9 6 15 12 9 18"
            stroke="#000"
            stroke-width="2"
          />
        </svg>
      </li>
    </ul>
  </main>
</div>
`;
